language/scripts/setup_env.py

```diff
@@ -138,13 +138,32 @@ def build_terra(terra_dir, llvm_dir, cache, is_cray, thread_count):
         cwd=terra_dir,
         env=env)
 
+def build_hdf(source_dir, install_dir, thread_count, is_cray):
+    env = None
+    if is_cray:
```

Comment:
Do we need this part? I think this was something we were doing for Terra because otherwise Terra wouldn't build properly, but I would expect HDF to build alright with the default system compiler.

Reply:
I'll try it on Titan and see if it works without this block.

Reply:
This block is necessary when building on Titan. During compilation of the HDF library, the build process needs to run an executable (H5Detect) that's produced at an earlier point, and without this block H5Detect contains instructions that are illegal for the head node.

Reply:
Fun.